### PR TITLE
Upgrade to Babel 7

### PR DESCRIPTION
**Pull request by ankurkaushal**
### Reasons for making this change

Using my work account (@ankurkaushal360), I opened an issue here, #1319 mentioning `babel` upgrade & I was told to it would be a good idea.

So this pull upgrades the Babel 6 to Babel 7 using `npx babel-upgrade --write` and does the following things:

- Removes deprecated `babel-plugin-react-transform` with `@babel/plugin-transform-react-jsx`
- Updates a few files with `default` exports as they were throwing warnings while running `dev` like this:

image

As far as I can tell, everything should work as it is. Even all the tests are passing too. I don't think I need to add / update any tests for this change. 

Also, I haven't updated the `package.json` version yet. Though I think I need to update it. Would you prefer `major` or `minor` version bump?

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [ ] **I'm adding or updating code**
  - [ ] I've added and/or updated tests
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [x] I've run `npm run cs-format` on my branch to conform my code to [prettier](https://github.com/prettier/prettier) coding style
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
   "plugins": [
-    "transform-object-rest-spread",
-    "transform-class-properties",
-    "transform-runtime"
+    "@babel/plugin-proposal-object-rest-spread",
+    "@babel/plugin-proposal-class-properties",
+    [
+      "@babel/plugin-transform-runtime",
+      {
+        "corejs": 2
```
Comment (epicfaace) [inline .babelrc]: Is there a reason why you chose 2 to be the value for corejs?

Comment (ankurkaushal) [reply .babelrc]: I think it was `npx babel-upgrade --write` that did it. And we're using `core-js` v2 right, so it should be that version right?

**Comment by ankurkaushal** [comment]
@epicfaace Marked this PR as 'ready for review' :)

**Comment by epicfaace** [comment]
Thanks!

**Pull request merged**